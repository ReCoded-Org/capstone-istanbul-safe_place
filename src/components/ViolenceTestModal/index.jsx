import React, { useState } from "react";
import { Row, Button, Modal } from "react-bootstrap";
import { Link } from "react-router-dom";
import { useTranslation } from "react-i18next";
import "./index.scss";

const ViolenceTestModal = () => {
  const { t } = useTranslation();
  const [isShown, setIsShown] = useState(false);

  const handleClose = () => setIsShown(false);
  const handleShow = () => setIsShown(true);

  return (
    <Row className="justify-content-md-center violenceModal">
      <Button variant="primary" onClick={handleShow}>
        {t("violenceTestPage.result")}
      </Button>

      <Modal show={isShown} onHide={handleClose} className="modalCard">
        <Modal.Body>
          <button type="button" className="close" onClick={handleClose}>
            <span aria-hidden="true">Ã—</span>
          </button>
          <div className="modalBody">
            <p>{t("violenceTestPage.testResult")}</p>
            <Button type="button">
              <a href="tel:+1-340-5558-235">340-5558-235</a>
            </Button>
            <p>{t("violenceTestPage.getHelp")}</p>
            <Link to="/seekhelp">
              <Button type="button">{t("violenceTestPage.seekHelpBtn")}</Button>
            </Link>
          </div>
        </Modal.Body>
      </Modal>
    </Row>
  );
};

export default ViolenceTestModal;
