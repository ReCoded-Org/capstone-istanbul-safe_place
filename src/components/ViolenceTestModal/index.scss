@import "../../style/variables";
@import "../../style/bootstrapImports";
@import "../../style/buttons";

.violenceModal {
  margin-bottom: 3rem;

}

.btn {
  @include primaryBtn;
  height:3em;
  width:12rem;

  &:active{
    background-color: $primaryColor!important;
    border-color: $primaryColor!important;
  }

  &:focus{
    background-color: $primaryColor;
    border-color:$primaryColor;
    box-shadow: 0 0 0 0.2rem $warningColor!important;
  }
}

.modal-content {
  display: flex !important;
  flex-direction: column!important;
  justify-content: baseline!important;
  align-content: center!important;
  align-items: !important;
}

.try {
  @include primaryBtn;
  height:3em;
  width:12rem;
}