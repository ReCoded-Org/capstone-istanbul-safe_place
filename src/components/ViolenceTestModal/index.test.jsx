import React from "react";
import renderer from "react-test-renderer";
import ViolenceTestModal from "./index";

it("renders correctly and matches the snapshot", () => {
  const tree = renderer.create(<ViolenceTestModal />).toJSON();
  expect(tree).toMatchSnapshot();
});
