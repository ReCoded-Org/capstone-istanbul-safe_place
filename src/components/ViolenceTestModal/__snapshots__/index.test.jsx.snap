// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly and matches the snapshot 1`] = `
<div
  className="justify-content-md-center violenceModal row"
>
  <button
    className="btn btn-primary"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    violenceTestPage.result
  </button>
</div>
`;
